frontend/src/components/ModelDiff.jsx

```diff
@@ -2,88 +2,104 @@
-    setIsExpanded(false);
+    const option = JSON.parse(e.target.value);
+    setSelectedOption(e.target.value);
+    setSelectedDetails(option);
+    setIsExpanded((prev) => ({ ...prev, details: false }));
   };
 
   const handleGenerateDiff = async () => {
-    if (!commitDetails[0].hash || !selectedDetails.hash) {
+    const headCommit = commitDetails[0]?.hash;
+    const selectedCommit = selectedDetails?.hash;
+
+    if (!headCommit || !selectedCommit) {
       alert('Please select a version.');
       return;
     }
-    setCompleteLoading(false);
-    setIsLoading(true);
+
+    setLoadingState('loading');
+    setError('');
+
     try {
-      const url = API_BASE_URL + '/compare';
-      const response = await postData(url, {
-        head: commitDetails[0].hash,
-        prev: selectedDetails.hash
+      const response = await postData(`${API_BASE_URL}/compare`, {
+        head: headCommit,
+        prev: selectedCommit
       });
       const data = await response.json();
+
       if (data.error) {
         throw new Error(data.error);
       }
+
+      setDiffSuccess(true);
+      setLoadingState('complete');
     } catch (error) {
-      console.error('Error:', error);
+      setDiffSuccess(false);
+      setError(error.message);
+      setLoadingState('error');
     } finally {
-      setIsLoading(false);
-      setCompleteLoading(true);
       onRefetch();
     }
   };
 
-  const postData = async (url = '', data = {}) => {
+  const postData = async (url, data) => {
     try {
       const response = await fetch(url, {
         method: 'POST',
-        headers: {
-          'Content-Type': 'application/json'
-        },
+        headers: { 'Content-Type': 'application/json' },
         body: JSON.stringify(data)
       });
-      if (!response.ok) {
-        throw new Error('Network response was not ok');
-      }
+
+      if (!response.ok) throw new Error('Network response was not ok');
       return response;
     } catch (error) {
       console.error('Error in postData:', error);
       throw error;
     }
   };
 
-  async function openModelCompareDialog() {
+  const openModelCompareDialog = async () => {
     try {
-      const response = await fetch(API_BASE_URL + '/commits');
-      if (!response.ok) {
-        throw new Error(
-          'Failed to fetch commits info: ' + response.statusText
-        );
-      }
+      setError('');
+      setSelectedOption('');
+      setSelectedDetails(null);
+      setDiffSuccess(null);
+      setLoadingState('idle');
+
+      const response = await fetch(`${API_BASE_URL}/commits`);
+
+      if (!response.ok) throw new Error('Failed to fetch commits info.');
+
       const data = await response.json();
       if (data === null) {
-        alert('No commits found.');
-        throw new Error('No commits found.');
+        alert('Not a git repo');
+        return;
+      } else if (data.length < 2) {
+        alert('Not enough commits to compare.');
+        return;
       }
```

Comment:
Seems we don't even get to this part, because the backend errors and responds with a 500.

```
  File "…/capella_model_explorer/backend/explorer.py", line 306, in get_commits
    result = model_diff.populate_commits(self.model)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "…/capella_model_explorer/backend/model_diff.py", line 122, in populate_commits
    commits = get_commit_hashes(path)
              ^^^^^^^^^^^^^^^^^^^^^^^
  File "…/capella_model_explorer/backend/model_diff.py", line 200, in get_commit_hashes
    commits = [_get_revision_info(path, c) for c in commit_hashes]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "…/capella_model_explorer/backend/model_diff.py", line 171, in _get_revision_info
    subject = description.splitlines()[0]
              ~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```